packages/autocomplete-core/src/onInput.ts

```diff
@@ -117,7 +126,24 @@ export function onInput<TItem extends BaseItem>({
             })
           )
             .then(resolve)
-            .then((responses) => postResolve(responses, sources, store))
+            .then((responses) => {
+              const __automaticInsights = responses.some(({ items }) =>
+                isSearchResponseWithAutomaticInsightsFlag(items)
```

Comment:
I see the method accepts a generic type, do you want to pass `TItem` here?

```suggestion
                isSearchResponseWithAutomaticInsightsFlag<TItem>(items)
```